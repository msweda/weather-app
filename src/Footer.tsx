import {
  copyright,
  email,
  emailLink,
  instagramLink,
  linkedInLink,
  location,
  phoneNumber,
} from '@/common/strings'
import theme from '@/common/theme'
import {
  createStyles,
  Theme,
  WithStyles,
  withStyles,
} from '@material-ui/core/styles'
import Typography from '@material-ui/core/Typography'
import EmailIcon from '@material-ui/icons/Email'
import PhoneIcon from '@material-ui/icons/Phone'
import PlaceIcon from '@material-ui/icons/Place'
import * as React from 'react'
import { SocialIcon } from 'react-social-icons'

const Footer: React.FC<Props> = ({ classes }) => (
  <div className={classes.footer}>
    <div className={classes.contact}>
      <div className={classes.iconWithText}>
        <PhoneIcon className={classes.icon} />
        <Typography>{phoneNumber}</Typography>
      </div>
      <div className={classes.iconWithText}>
        <EmailIcon className={classes.icon} />
        <Typography>{email}</Typography>
      </div>
      <div className={classes.iconWithText}>
        <PlaceIcon className={classes.icon} />
        <Typography>{location}</Typography>
      </div>
    </div>
    <Typography className={classes.copyright}>
      {copyright(Date.now())}
    </Typography>
    <div className={classes.socialMediaIcons}>
      <SocialIcon url={emailLink} bgColor={theme.palette.secondary.main} />
      <SocialIcon url={linkedInLink} bgColor={theme.palette.secondary.main} />
      <SocialIcon url={instagramLink} bgColor={theme.palette.secondary.main} />
    </div>
  </div>
)

const styles = (theme: Theme) =>
  createStyles({
    footer: {
      height: 100,
      display: 'flex',
      flexDirection: 'row',
      justifyContent: 'space-around',
      alignItems: 'center',
      position: 'relative',
      [theme.breakpoints.down('xs')]: {
        height: 200,
        flexDirection: 'column',
        alignItems: 'stretch',
      },
    },
    contact: {
      display: 'flex',
      flexDirection: 'column',
      justifyContent: 'flex-start',
      alignItems: 'stretch',
      [theme.breakpoints.down('xs')]: {
        alignItems: 'center',
      },
      '& > *': {
        [theme.breakpoints.down('xs')]: {
          textAlign: 'center',
        },
      },
    },
    iconWithText: {
      display: 'flex',
      flexDirection: 'row',
      justifyContent: 'flex-start',
      alignItems: 'center',
      marginBottom: 5,
    },
    icon: {
      color: theme.palette.text.primary,
      marginRight: 10,
      fontSize: 14,
    },
    copyright: {
      marginBottom: 5,
      [theme.breakpoints.down('xs')]: {
        textAlign: 'center',
      },
    },
    socialMediaIcons: {
      display: 'flex',
      flexDirection: 'row',
      justifyContent: 'center',
      alignItems: 'center',
      '& > *': {
        margin: 5,
      },
    },
  })

interface Props extends WithStyles<typeof styles> {}

export default withStyles(styles)(Footer)
