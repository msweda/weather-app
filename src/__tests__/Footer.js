import {
  copyright as expectedCopyright,
  email as expectedEmail,
  emailLink as expectedEmailLink,
  instagramLink as expectedInstagramLink,
  linkedInLink as expectedLinkedInLink,
  location as expectedLocation,
  phoneNumber as expectedPhoneNumber,
} from '@/common/strings'
import { createShallow } from '@material-ui/core/test-utils'
import Typography from '@material-ui/core/Typography'
import { advanceTo, clear } from 'jest-date-mock'
import * as React from 'react'
import { SocialIcon } from 'react-social-icons'
import Footer from '../Footer'

describe('<Footer />', () => {
  let shallow
  let wrapper

  beforeEach(() => {
    shallow = createShallow({ untilSelector: Footer })
    wrapper = shallow(<Footer />)
  })

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should display my phone number', () => {
    const phoneNumber = wrapper
      .find(Typography)
      .at(0)
      .children()
      .text()
    expect(phoneNumber).toEqual(expectedPhoneNumber)
  })

  it('should display my email', () => {
    const email = wrapper
      .find(Typography)
      .at(1)
      .children()
      .text()
    expect(email).toEqual(expectedEmail)
  })

  it('should display my location', () => {
    const location = wrapper
      .find(Typography)
      .at(2)
      .children()
      .text()
    expect(location).toEqual(expectedLocation)
  })

  it('should display a copyright for the current year', () => {
    const mockDate2018 = new Date(2018)
    advanceTo(mockDate2018)
    wrapper = shallow(<Footer />)
    const copyright2018 = wrapper
      .find(Typography)
      .at(3)
      .children()
      .text()
    expect(copyright2018).toEqual(expectedCopyright(mockDate2018))

    const mockDate2020 = new Date(2020)
    advanceTo(mockDate2020)
    wrapper = shallow(<Footer />)
    const copyright2020 = wrapper
      .find(Typography)
      .at(3)
      .children()
      .text()
    expect(copyright2020).toEqual(expectedCopyright(mockDate2020))

    clear()
  })

  it('should display an icon with an email link', () => {
    const socialIconEmail = wrapper.find(SocialIcon).at(0)
    expect(socialIconEmail.props().url).toEqual(expectedEmailLink)
  })

  it('should display an icon with a linkedIn link', () => {
    const socialIconEmail = wrapper.find(SocialIcon).at(1)
    expect(socialIconEmail.props().url).toEqual(expectedLinkedInLink)
  })

  it('should display an icon with an instagram link', () => {
    const socialIconEmail = wrapper.find(SocialIcon).at(2)
    expect(socialIconEmail.props().url).toEqual(expectedInstagramLink)
  })
})
