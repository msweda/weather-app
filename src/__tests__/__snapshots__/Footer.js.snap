// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should match snapshot 1`] = `
<div
  className="Footer-footer-1"
>
  <div
    className="Footer-contact-2"
  >
    <div
      className="Footer-iconWithText-3"
    >
      <pure(PhoneIcon)
        className="Footer-icon-4"
      />
      <WithStyles(Typography)>
        (248)-770-7487
      </WithStyles(Typography)>
    </div>
    <div
      className="Footer-iconWithText-3"
    >
      <pure(EmailIcon)
        className="Footer-icon-4"
      />
      <WithStyles(Typography)>
        msweda.developer@gmail.com
      </WithStyles(Typography)>
    </div>
    <div
      className="Footer-iconWithText-3"
    >
      <pure(PlaceIcon)
        className="Footer-icon-4"
      />
      <WithStyles(Typography)>
        Detroit, MI
      </WithStyles(Typography)>
    </div>
  </div>
  <WithStyles(Typography)
    className="Footer-copyright-5"
  >
    Â© Matthew Sweda 2019
  </WithStyles(Typography)>
  <div
    className="Footer-socialMediaIcons-6"
  >
    <SocialIcon
      bgColor="#6ba099"
      url="mailto:msweda.developer@gmail.com"
    />
    <SocialIcon
      bgColor="#6ba099"
      url="https://www.linkedin.com/in/matthew-sweda-118570ba/"
    />
    <SocialIcon
      bgColor="#6ba099"
      url="https://www.instagram.com/mattsweda/"
    />
  </div>
</div>
`;
